src/services/outliningElementsCollector.ts

```diff
@@ -84,6 +85,7 @@ namespace ts.OutliningElementsCollector {
         let depth = 0;
         const maxDepth = 20;
         function walk(n: Node): void {
+            cancellationToken.throwIfCancellationRequested();

```

Comment:
I've removed the throttling until I can prove it is required as it adds overhead to deal with test failures.  Will confirm it is required before adding it back in.